### PR TITLE
Improve cache interface for CustomResourceEventSource

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java [hunk 1]
```diff
@@ -176,7 +179,19 @@ public void onDelete(T resource, boolean b) {
     }
   }
 
+  public Stream<T> getCachedCustomResources() {
+    return getCachedCustomResources(a -> true);
+  }
 
+  public Stream<T> getCachedCustomResources(Predicate<T> predicate) {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java:186]: That seems like a very inefficient method.

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java:186]: Changed it to a way how it's doable nicely with current java APIs but not sure it's actually much more efficient;

**Comment by metacosm** [comment]
Should we close this?

**Comment by csviri** [comment]
> Should we close this?

or merge it? :)

**Review by metacosm — APPROVED** [review]

**Pull request merged**